examples/simple_plugins/plugins/decorator.yaml

```diff
@@ -1,3 +1,4 @@
+id: decorator
```

Comment:
So far we've relied purely on file naming as a way to identify plugins. This change might complicate that.

For example, what happens if this value does not match the file name? I'm concerned that this could lead to inconsistencies.

Reply:
I can't think of a reason that we need to care if it matches the file name. I think it still makes sense to make it a convention that they match, but I can't think of a reason it is necessary. 

Mostly, my thought process here is, if ID is not part of the definition, where do we keep it when we have a representation of a plugin that is not just a file? We were kinda hacking around this before because we never had to marshal a plugin, so we could just unmarshal the plugin, then set the ID as the name of the file. 

What do you think about making it backwards compatible such that, if `id` is not specified in the plugin definition, we default to file name? That way we can get plugins from other sources that aren't files (like through the websocket).

Reply:
Our previous decision to rely on filename was largely a usability decision. Implementation details aside, I think that what we currently have is both elegant and intuitive, and that this change would be a regression in that regard. For example:
- Do I have this plugin installed? If the file exists, yes. If not, no.
- Can I have the same plugin installed twice? No

> I can't think of a reason that we need to care if it matches the file name

We need to care because users are affected by this. Our most important heuristic is to ensure a smooth path to adoption. Usability regressions are very important and should be avoided if possible.

> if ID is not part of the definition, where do we keep it when we have a representation of a plugin that is not just a file

This is a fair point. However, I think there are answers to this other than what you've proposed.

For example, `id` can remain a private field and be used by stanza exactly as it has been.
```
type Definition struct {
	id               string
	Version          string               `json:"version"     yaml:"version"`
	Title            string               `json:"title"       yaml:"title"`
	Description      string               `json:"description" yaml:"description"`
	Parameters       map[string]Parameter `json:"parameters"  yaml:"parameters"`
	MinStanzaVersion *string              `json:"minStanzaVersion" yaml:"min_stanza_version"`
	MaxStanzaVersion *string              `json:"minStanzaVerion" yaml:"max_stanza_version"`
}
```

For the "meta" use cases that we're trying to solve with this change, we can pass `id` alongside the definition, rather than inside of it.

```
{
    "id": "zookeeper",
    "definition": {
        "version": "0.0.1",
        "title": "Apache Zookeeper",
        ...
```

Rather than:
```
{
    "definition": {
        "id": "zookeeper",
        "version": "0.0.1",
        "title": "Apache Zookeeper",
        ...
```

Do you think this pattern would be acceptable?

Reply:
Thanks for the detailed response. You're right about the change in user expectations -- I hadn't thought that fully through. I think your proposal of moving ID out of the definition makes sense. I'll take a pass at implementing that and update the PR